app/server/utils/metrics/metric.py

```diff
@@ -6,63 +6,122 @@
+        :param population_query_result: This is a representation of the number of users over time, used in 
+            post-processing of certian metrics. See process_timeseries.py for more details.
+        :param calculate_only_aggregates: if true, this skips calculating timeseries data and only fetches
+             aggregated_query and total_query
+        """
+        actions = { 'query': self.query_actions, 'aggregated_query': self.aggregated_query_actions, 'total_query': self.total_query_actions }
+        if self.is_timeseries:
+            if calculate_only_aggregates:
+                queries = { 'aggregated_query': self.aggregated_query, 'total_query': self.total_query }
+            else:   
+                queries = { 'query': self.query, 'aggregated_query': self.aggregated_query, 'total_query': self.total_query }
+           
+            if None in queries.values():
+                raise Exception('Timeseries query requires a query, aggregated_query, and a total_query')
+        else:
+            queries = { 'query': self.query }
 
-        # Validate that the filters we're applying are in the metrics' filterable_by
-        for f in user_filters or []:
-            if f not in self.filterable_by:
-                raise Exception(f'{self.metric_name} not filterable by {f}')
+        results = {}
+        for query in queries:
+            user_filters = user_filters or {}
+            # Apply stock filters
+            filtered_query = queries[query]
+            for f in self.stock_filters:
+                filtered_query = filtered_query.filter(*f)
 
-        if DATE in self.filterable_by or []:
-            filtered_query = filtered_query.filter(*date_filters)
+            # Validate that the filters we're applying are in the metrics' filterable_by
+            for f in user_filters or []:
+                if f not in self.filterable_by:
+                    raise Exception(f'{self.metric_name} not filterable by {f}')
+            # Apply the applicable date filters
+            if DATE in self.filterable_by or []:
+                if date_filters_dict:
+                    date_filters = date_filters_dict[self.object_model]
+                    filtered_query = filtered_query.filter(*date_filters)
 
-        if not self.bypass_user_filters:
-            filtered_query = filters.apply_filters(filtered_query, user_filters, self.object_model)
+            if not self.bypass_user_filters:
+                filtered_query = filters.apply_filters(filtered_query, user_filters, self.object_model)
 
-        result = metrics_cache.execute_with_partial_history_cache(self.metric_name, filtered_query, self.object_model, self.caching_combinatory_strategy, enable_caching) 
-        if not self.timeseries_actions:
+            result = metrics_cache.execute_with_partial_history_cache(self.metric_name, filtered_query, self.object_model, self.caching_combinatory_strategy, enable_caching) 
```

Comment:
nit: Can we try keep line lengths to <120 chars? (few places like this) black.py is a todo!